src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -15,7 +15,7 @@ namespace Microsoft.CodeAnalysis
     /// Represents a diagnostic, such as a compiler error or a warning, along with the location where it occurred.
     /// </summary>
     [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
-    public abstract partial class Diagnostic : IEquatable<Diagnostic>, IFormattable
```

Comment:
@cston Consider adding information about Equals behavior to the doc comment for the class.